### PR TITLE
Add support for GO on z/OS.

### PR DESCRIPTION
**Pull request by jbyibm**
This implementation add support for z/OS GO.
See: https://www.ibm.com/products/open-enterprise-sdk-go-zos.

### CODE REVIEW

**Comment by jbyibm** [comment]
This PR is the implementation of the issue #33

File: extractor_zos.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+func lchtimes(name string, mode os.FileMode, atime, mtime time.Time) error {
+    err := lchmod(name, mode)
+    if err != nil {
+        return err
+    }
+    err = os.Chtimes(name, atime, mtime)
+    if err != nil {
+        return &os.PathError{Op: "lchtimes", Path: name, Err: err}
+    }
+    return nil
```
Comment (saracen) [inline extractor_zos.go:30]: I'm assuming that `lutimes` is supported by `zos`, but it's available as part of `golang.org/x/sys/unix` yet?

I wonder if we can support it by calling the syscall directly. Are there any plans for `zos` support to be added to `golang.org/x/sys/unix`?

**Review by saracen — COMMENTED** [review]
Thank you @jbyibm 

I'm confused by https://www.ibm.com/products/open-enterprise-sdk-go-zos - is the zos Go compiler open?

Is it possible for me to cross compile and test in some emulated environment?

**Comment by jbyibm** [comment]
> Thank you @jbyibm
> 
> I'm confused by https://www.ibm.com/products/open-enterprise-sdk-go-zos - is the zos Go compiler open?
> 
> Is it possible for me to cross compile and test in some emulated environment?

No it is not possible to cross compile. You need a real native z/OS or an emulated z/OS with zD&T (https://www.ibm.com/docs/fr/zdt/13.3.0)

File: extractor_zos.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    if mode&os.ModeSymlink != 0 {
+        // it is a symlink, do not follow
+        return nil
+    }
+    err := os.Chmod(name, mode)
+    if err != nil {
+        return &os.PathError{Op: "lchmod", Path: name, Err: err}
+    }
+    return nil
+}
```
[the remark this answers is not in the document]
Comment (jbyibm) [reply extractor_zos.go:15]: Thanks for your comment. I just updated the new version.

File: extractor_zos.go (hunk 1 above)
Comment (jbyibm) [reply extractor_zos.go:26]: I need to check internally.

Comment (jbyibm) [reply extractor_zos.go:26]: `lutimes` is not yet available natively on zos < 2.5. The implementation here is a workaround. We have tested this internally with fastzip APIs.

**Comment by saracen** [comment]
> No it is not possible to cross compile. You need a real native z/OS or an emulated z/OS with zD&T (https://www.ibm.com/docs/fr/zdt/13.3.0)

@jbyibm Do I need a license for this? It'll be difficult to make changes to this library in the future that might affect `zos` if it's not part of the standard Go compiler and there's no way to check it at least compiles.

Is there something like a dockerized version of zD&T available for open-source CI that can be used without a license? Thank you.

**Comment by saracen** [comment]
@jbyibm I'm going to close this for now. However, please do let me know if it'll eventually be possible for open-source developers to test against the system they'll be targeting.

**Pull request closed without merging**